src/layout/Box.js

```diff
@@ -0,0 +1,410 @@
+
+const getSpaceValue = (
+  property: string,
+  theme: Object,
+  value: number | string
+) => theme[`space_${spacingTypes[property]}_${value}`] || getMeasurement(value);
+
+const getSpacingStyles = (
+  property: string,
+  restProps: Object,
+  rtl: boolean
+): SpacingStyles => {
+  // prettier-ignore
+  const spacingPropKeys = ['', 'Horizontal', 'Vertical', 'Start', 'End', 'Bottom', 'Left', 'Right', 'Top']
+    .map(suffix => `${property}${suffix}`)
+    .filter(key => restProps[key] !== undefined);
+
+  const setStyles = (
+    directions: Array<string>,
+    key: string,
+    styles: SpacingStyles
+  ) => {
+    directions.forEach(direction => {
+      styles[`${property}${direction}`] = restProps[key];
+    });
+  };
+
+  return spacingPropKeys.reduce((acc, key) => {
+    if (key === property) {
+      setStyles(['Bottom', 'Left', 'Right', 'Top'], key, acc);
+    } else if (key.indexOf('Horizontal') !== -1) {
+      setStyles(['Left', 'Right'], key, acc);
+    } else if (key.indexOf('Vertical') !== -1) {
+      setStyles(['Bottom', 'Top'], key, acc);
+    } else if (key.indexOf('Start') !== -1) {
+      setStyles([`${rtl ? 'Right' : 'Left'}`], key, acc);
+    } else if (key.indexOf('End') !== -1) {
+      setStyles([`${rtl ? 'Left' : 'Right'}`], key, acc);
+    } else {
+      acc[key] = restProps[key];
+    }
+    return acc;
+  }, {});
+};
+
+const styles = {
+  root: ({ breakpoints, height, inline, theme, width, ...restProps }) => {
+    const rtl = theme.direction === 'rtl';
+
+    const mapValueToProperty = (
+      property: string,
+      value: Values
+    ): number | string => {
+      const map = {
+        display: value => (value ? 'inline-block' : 'block'),
+        height: getMeasurement,
+        width: getMeasurement,
+        ...['margin', 'padding'].reduce((acc, property) => {
+          Object.keys(
+            getSpacingStyles(property, restProps, rtl)
+          ).forEach(style => {
+            acc[style] = value => getSpaceValue(property, theme, value);
+          });
+          return acc;
+        }, {})
+      };
+
+      return map[property](value);
+    };
+
+    return getResponsiveStyles({
+      breakpoints,
+      mapValueToProperty,
+      styles: {
+        display: inline,
+        height,
+        ...getSpacingStyles('margin', restProps, rtl),
+        ...getSpacingStyles('padding', restProps, rtl),
+        width
+      },
+      theme
+    });
+  }
+};
+
+// Box's root node must be created outside of render, so that the entire DOM
+// element is replaced only when the element prop is changed, otherwise it is
+// updated in place
+function createRootNode(props: Props) {
+  const { element = Box.defaultProps.element } = props;
+
+  return createStyledComponent(element, styles.root, {
+    includeStyleReset: true,
+    rootEl: element
+  });
+}
+
+/**
+ * Box component provides an easy way to apply standardized size & space to your
+ * layout.
```

Comment:
Since Box is the entry point for Layout, should this description have links to the other Layout components?

Reply:
Probably a good idea, as the people I've shared this with have had a hard time noticing the multiple components.

Reply:
I really appreciate you getting outside feedback on this!